ft_preprocessing.m

```diff
@@ -364,25 +364,34 @@
     [dataout.trial{i}, dataout.label, dataout.time{i}, cfg] = preproc(data.trial{i}, data.label,  data.time{i}, cfg, begpadding, endpadding);
     
   end % for all trials
-  
-  if isstruct(cfg.montage) && strcmp(cfg.updatesens, 'yes')
-    % apply the linear projection also to the sensor description
-    if issubfield(cfg.montage, 'type')
-      bname = cfg.montage.type;
-    else
-      bname = 'preproc';
-    end
-    if isfield(dataout, 'grad')
-      fprintf('applying the montage to the grad structure\n');
-      dataout.grad = ft_apply_montage(dataout.grad, cfg.montage, 'feedback', 'none', 'keepunused', 'yes', 'balancename', bname);
-    end
-    if isfield(dataout, 'elec')
-      fprintf('applying the montage to the grad structure\n');
-      dataout.elec = ft_apply_montage(dataout.elec, cfg.montage, 'feedback', 'none', 'keepunused', 'yes', 'balancename', bname);
-    end
-    if isfield(dataout, 'opto')
-      fprintf('applying the montage to the opto structure\n');
-      dataout.opto = ft_apply_montage(dataout.opto, cfg.montage, 'feedback', 'none', 'keepunused', 'yes', 'balancename', bname);
+    
+  if strcmp(cfg.updatesens, 'yes') && (isfield(dataout, 'grad') || isfield(dataout, 'elec') || isfield(dataout, 'opto'))
+    if strcmp(cfg.reref, 'yes') && ~isstruct(cfg.montage)
+      fprintf('creating an identity matrix montage');
```

Comment:
why would this be the montage that is to be applied. The output cfg will now be incorrect and not describe the operations that are done on the output data. 

Please note that 

cfg = ...
data2 = ft_preprocessing(cfg, data1)
data3 = ft_preprocessing(data2.cfg, data1)
assert(data2==data3)

should hold.